### PR TITLE
Adding vpc endpoint to bedrock client builder

### LINKED ISSUE #1846: [FEATURE] add vpce bedrock capabilities
**Is your feature request related to a problem? Please describe.**
I want to use langchain4j for bedrock with private link, i had to create custom classes to do that cause client creation with url is private method.

**Describe the solution you'd like**
I want to have bean urlOverride capabilities when i use private link to connecto to bedrock (ex : https://vpce...***)

**Describe alternatives you've considered**
Alternatively can inject the URL by env or application prop

**Additional context**
Actually we have to create CustomAbstractAwsClient with different getClient() method cause its private. If this method was protected, we can just override this by extension. 

Ty :)

### PR DESCRIPTION
**Pull request by rndtavares**
## Issue
Closes #1846 

## Change
I added the endpointOverride field to AbstractSharedBedrockChatModel and used it in AbstractBedrockStreamingChatModel to create the BedrockRuntimeAsyncClient and to AbstractBedrockChatModel to create the BedrockRuntimeClient client, also added the same field to AbstractBedrockEmbeddingModel and used it to create the BedrockRuntimeClient.

## General checklist

- [X] There are no breaking changes
- [X] I have added unit and integration tests for my change 
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
@rndtavares thank you! Could you test it please?

**Comment by rndtavares** [comment]
Tests ok
image

**Comment by rndtavares** [comment]
can you review @langchain4j?

**Comment by langchain4j** [comment]
@rndtavares sorry for delays, I will try to look into this PR during the week. Thank you for your patience!

File: langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockChatModelIT.java [hunk 1]
```diff
@@ -773,6 +774,31 @@ void testBedrockMistralAiMixtral8x7bInstructChatModel() {
...
 
+    @Test
+    void testBedrockAnthropicV3_5SonnetChatModelWithVPCe() {
+
+        BedrockAnthropicMessageChatModel bedrockChatModel = BedrockAnthropicMessageChatModel
+                .builder()
+                .temperature(0.50f)
+                .maxTokens(300)
+                .region(Region.US_EAST_1)
+                .model(BedrockAnthropicMessageChatModel.Types.AnthropicClaude3_5SonnetV1.getValue())
```
Comment (langchain4j) [inline langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockChatModelIT.java]: minor: can we use cheaper model for this test please?

**Review by ghost — COMMENTED** [review]
@rndtavares thank you! Looks good, please change the model and we can merge

File: langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockChatModelIT.java (hunk 1 above)
Comment (rndtavares) [reply langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockChatModelIT.java]: done

**Comment by langchain4j** [comment]
Hey @rndtavares, there is a https://github.com/langchain4j/langchain4j/pull/2285 which allows setting the client (including the endpoint override). Considering this, do you think we need a dedicated setter for endpoint override (this PR)?

**Comment by rndtavares** [comment]
> #2285

This is what I need, if I can pass the BedrockRuntimeClient it is ok, no need for this PR

**Comment by langchain4j** [comment]
Closing as this will be possible with https://github.com/langchain4j/langchain4j/pull/2285

**Pull request closed without merging**